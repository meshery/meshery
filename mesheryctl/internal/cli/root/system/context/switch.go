package context

import (
	"errors"

	log "github.com/sirupsen/logrus"

	"github.com/spf13/cobra"
	"github.com/spf13/viper"
)

// switchContextCmd represents the switch command
var switchContextCmd = &cobra.Command{
	Use:          "switch <context-name>",
	Short:        "switch context",
	Long:         `Configure mesheryctl to actively use one one context vs. the another context`,
	Args:         cobra.ExactArgs(1),
	SilenceUsage: true,
	RunE: func(cmd *cobra.Command, args []string) error {
		err := viper.Unmarshal(&configuration)
		if err != nil {
			return err
		}
		_, exists := configuration.Contexts[args[0]]
		if !exists {
			return errors.New("requested context does not exist")
		}
		if viper.GetString("current-context") == args[0] {
			return errors.New("already using context '" + args[0] + "'")
		}
		configuration.CurrentContext = args[0]
		viper.Set("current-context", configuration.CurrentContext)
		log.Printf("switched to context '%s'", args[0])
		err = viper.WriteConfig()
		return err
	},
}
