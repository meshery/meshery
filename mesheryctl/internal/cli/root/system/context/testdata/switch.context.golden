switched to context 'local2'
